core/src/main/java/com/amplifyframework/predictions/models/Polygon.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.predictions.models;
+
+import android.graphics.Point;
+import androidx.annotation.NonNull;
+
+import com.amplifyframework.util.Immutable;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A class to represent more complex shapes than a simple
+ * rectangular box.
+ */
+public final class Polygon {
+    private static final int MINIMUM_POINTS_REQUIRED = 3;
+
+    private final List<Point> points;
+
+    private Polygon(List<Point> points) {
+        this.points = points;
+    }
+
+    /**
+     * Gets the polygon vertices.
+     * @return the polygon vertices
+     */
+    @NonNull
+    public List<Point> getPoints() {
+        return Immutable.of(points);
+    }
+
+    /**
+     * Construct a new immutable instance of polygon from a
+     * list of points.
+     * @param points a list of points representing the corners
+     *               of a polygon object
+     * @return A polygon instance
+     */
+    @NonNull
+    public static Polygon fromPoints(@NonNull List<Point> points) {
+        Objects.requireNonNull(points);
+        if (points.size() < MINIMUM_POINTS_REQUIRED) {
+            throw new IllegalArgumentException("A polygon must " +
+                    "contain at least three points.");
+        }
+        return new Polygon(points);
```

Comment:
I don't think we need to validate this, in code, here. But, as a math thought for the day - is do the following coordinates in 3-space define a "polygon"? [0,1,0], [0,3,0], [0,2,0]?

Apparently!

https://gamedev.stackexchange.com/questions/34225/can-a-straight-line-be-called-a-polygon

Reply:
haha i was thinking this exact thing when I wrote this